src/include/storage/copier/compression.h

```diff
@@ -0,0 +1,219 @@
+#pragma once
+
+#include <cstdint>
+#include <limits>
+
+#include "common/types/types.h"
+
+namespace kuzu {
+namespace storage {
+
+// Returns the size of the data type in bytes
+uint32_t getDataTypeSizeInChunk(const common::LogicalType& dataType);
+
+// Compression type is written to the data header both so we can usually catch issues when we
+// decompress uncompressed data by mistake, and to allow for runtime-configurable compression in
+// future.
+enum class CompressionType : uint8_t {
+    COPY = 0,
+    INTEGER_BITPACKING = 1,
+    BOOLEAN = 2,
+};
+
+struct CompressionMetadata {
+    uint64_t numValuesPerPage;
```

Comment:
> I guess it could instead calculate the number of values per page based on the type and the type of compression used via a CompressionMetadata function.

Finished this part